apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx

```diff
@@ -14,44 +15,75 @@ import { Address } from "#/subgraph/src/scripts/last-addr";
 import { getIpfsMetadata } from "@/utils/ipfsUtils";
 import { pointSystems, poolTypes } from "@/types";
 import { CV_SCALE_PRECISION } from "@/utils/numbers";
+import { useEffect, useMemo, useState } from "react";
+import LoadingSpinner from "@/components/LoadingSpinner";
+import useSubgraphQueryByChain from "@/hooks/useSubgraphQueryByChain";
 
 export const dynamic = "force-dynamic";
 
 export type AlloQuery = getAlloQuery["allos"][number];
 
-const { urqlClient } = initUrqlClient();
-
-export default async function Pool({
+export default function Pool({
   params: { chain, poolId, garden },
 }: {
   params: { chain: string; poolId: number; garden: string };
 }) {
-  const { data } = await queryByChain<getPoolDataQuery>(
-    urqlClient,
+  const { data, fetching, error } = useSubgraphQueryByChain<getPoolDataQuery>(
     chain,
     getPoolDataDocument,
     { poolId: poolId, garden: garden },
+    {},
+    {
+      topic: "pool",
+      id: poolId,
+      chainId: chain,
+    },
   );
-  const strategyObj = data?.cvstrategies?.[0] as CVStrategy;
-  //const { tooltipMessage, isConnected, missmatchUrl } = useDisableButtons();
 
-  if (!strategyObj) {
-    return <div>{`Pool ${poolId} not found`}</div>;
+  useEffect(() => {
+    if (error) {
+      console.error("Error while fetching pool data: ", error);
+    }
+  }, [error]);
+
+  const [ipfsResult, setIpfsResult] =
+    useState<Awaited<ReturnType<typeof getIpfsMetadata>>>();
+
+  const metadata = data?.cvstrategies?.[0]?.metadata as string;
+
+  useEffect(() => {
+    if (metadata) {
+      getIpfsMetadata(metadata).then((data) => {
+        setIpfsResult(data);
+      });
+    }
+  }, [metadata]);
+
+  if (fetching || !ipfsResult) {
+    return (
+      <div className="mt-96">
+        <LoadingSpinner />
+      </div>
+    );
   }
 
-  const pointSystem = data?.cvstrategies?.[0].config?.pointSystem;
+  const strategyObj = data?.cvstrategies?.[0] as CVStrategy;
```

Comment:
still have wrong type, should be the same as the query to avoid undefined/null values if the subgraph changes.

Reply:
Looks like its not the same type exactly, its a subset of CVStrategy so I still need to cast it but I can add the undefined to the typecast:
![alt text](image.png)

Reply:
![image](https://github.com/1Hive/gardens-v2/assets/13890618/0574fc24-4b1b-4c26-beb5-aa8a887565f0)

Reply:
you just need extract the type, we have some example i did in the main page with it getTokenGardensQuery if not mistake.

Reply:
https://github.com/1Hive/gardens-v2/pull/229/commits/cba84b27bc34d75ea052c02f470504fa0d2ff897

Reply:
https://github.com/1Hive/gardens-v2/pull/229/commits/b21d5df3735c22b65ad6db7ae0ff2b39ef75dae4

Reply:
https://github.com/1Hive/gardens-v2/pull/229/commits/e4ad8dce7bc59079d1ba2ebb5496a792b591224d

Reply:
https://github.com/1Hive/gardens-v2/pull/229/commits/e4ad8dce7bc59079d1ba2ebb5496a792b591224d